src/Table.js

```diff
@@ -0,0 +1,189 @@
+    {category: "Electronics", price: "$399.99", stocked: false, name: "iPhone 5"},
+    {category: "Electronics", price: "$199.99", stocked: true, name: "Nexus 7"},
+    {category: "Electric", price: "$9.99", stocked: true, name: "Cable"},
+    {category: "Electric", price: "$2.99", stocked: false, name: "Socket"},
+    {category: "Electric", price: "$5.99", stocked: true, name: "Plug"},
+    {category: "Electric", price: "$8.99", stocked: true, name: "Lamp"},
+  ];
+
+function ProductRow(props) {
+    const stocked = props.productStocked;
+    const productNameClass = stocked ? "product-name" : "product-name-red";
+    return (
+        <div className="row">
+            <div className={productNameClass}>{props.productName}</div>
+            <div className="product-price">{props.productPrice}</div>
+        </div>
+    );
+}
+
+function ProductCategoryRow(props) {
+    return (
+        <div className="category-row">
+            {props.category}
+        </div>
+    );
+}
+
+function ProductTable(props) {
+    const filterText = props.filterText;
+    const inStockOnly = props.inStockOnly;
+    const table = [...data];
+    const filteredTable = filterTable(table, filterText, inStockOnly);
+    //filteredTable.map((v) => console.log("ProductTable.name: " + v.name));
+    //console.log("Size:  " + filteredTable.length);  
+    return (
+        <div className="table">
+            <div className="row">
+                <div className="product-desc">Name</div>
+                <div className="product-desc">Price</div>
+            </div>
+            <div>
+                {filteredTable.map((v, i, tab) => {
+                    if(i === 0) {
+                        return (
+                            <Fragment key={v.name} >
+                                <ProductCategoryRow category={v.category} />
+                                <ProductRow productName={v.name} productPrice={v.price} productStocked={v.stocked} />
+                            </Fragment>
+                        );
+                    } else if (i > 0 && (v.category !== tab[i - 1].category)) {
+                        return (
+                            <Fragment key={v.name} >
+                                <ProductCategoryRow category={v.category} />
+                                <ProductRow productName={v.name} productPrice={v.price} productStocked={v.stocked} />
+                            </Fragment>
+                        );
+                    } else {
+                        return (
+                            <Fragment key={v.name} >
+                                <ProductRow productName={v.name} productPrice={v.price} productStocked={v.stocked} />
+                            </Fragment>
+                        );
+                    }
+                })}
+            </div>
+        </div>
+    );
+}
+
+function SearchBar(props) {
+    const filterText = props.filterText;
+    const inStockOnly = props.inStockOnly;
```

Comment:
to jest spoko ale proponuję dekompozycję 
`const {filterText, inStockOnly} = props`